airbyte-integrations/connectors/source-exchange-rates/source_exchange_rate/source.py

```diff
@@ -0,0 +1,63 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from typing import Optional, Mapping
+
+import requests
+
+from base_python import AbstractSource, HttpStream, Stream
+from typing import Any, Tuple, List, Iterable
+
+class ExchangeRates(HttpStream):
+    url_base = "https://api.ratesapi.io/"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def path(self, **kwargs) -> str:
+        return "api/latest"
```

Comment:
Maybe we shouldn't query for latest in this connector but use the date instead. If we do so then the date can be our state.

Reply:
yup, we can also use stream_state to determine what the date should be. I think this is also a better fit for batching. We should create this PR once #2924 is merged (or rebase it and build on top of it)

Reply:
yup! was slowly inching my way through python and santa land to this end state.